sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/screens/home/HomeViewModel.kt

```diff
@@ -12,26 +13,51 @@ import javax.inject.Inject
 
 @HiltViewModel
 class HomeViewModel @Inject constructor(
-    private val useCase: UseCase,
+    private val getModelsUseCase: GetModelsUseCase,
+    private val getFirstTimeLaunchPreferencesUseCase: GetFirstTimeLaunchPreferencesUseCase,
+    private val updateFirstTimeLaunchPreferencesUseCase: UpdateFirstTimeLaunchPreferencesUseCase,
     dispatchers: DispatchersProvider
 ) : BaseViewModel(dispatchers) {
 
     private val _uiModels = MutableStateFlow<List<UiModel>>(emptyList())
-    val uiModels: StateFlow<List<UiModel>>
-        get() = _uiModels
+
+    private val _firstTimeLaunch = MutableStateFlow(false)
+
+    val homeViewState: StateFlow<HomeViewState> = combine(
+        _uiModels,
+        _firstTimeLaunch,
+        showLoading
+    ) { uiModels, firstTimeLaunch, showLoading ->
+        HomeViewState(
+            uiModels = uiModels,
+            firstTimeLaunch = firstTimeLaunch,
+            showLoading = showLoading
+        )
+    }.stateIn(
+        scope = viewModelScope,
+        started = SharingStarted.WhileSubscribed(5000),
+        initialValue = HomeViewState()
+    )
 
     init {
         execute {
-            showLoading()
-            useCase()
-                .catch {
-                    _error.emit(it)
-                }
-                .collect { result ->
-                    val uiModels = result.toUiModels()
-                    _uiModels.emit(uiModels)
+            val getModelsFlow = getModelsUseCase()
+            val getFirstTimeLaunchPreferencesFlow = getFirstTimeLaunchPreferencesUseCase()
+
+            getModelsFlow.combine(getFirstTimeLaunchPreferencesFlow) { uiModels, firstTimeLaunch ->
+                _uiModels.emit(uiModels.toUiModels())
+
+                _firstTimeLaunch.emit(firstTimeLaunch)
+                if (firstTimeLaunch) {
+                    updateFirstTimeLaunchPreferencesUseCase(false)
```

Comment:
What about if we return `Unit` instead of `Boolean`? Then we can get rid of the condition in `HomeScreen` inside `LaunchedEffect` scope.
```suggestion
                if (firstTimeLaunch) {
                 _firstTimeLaunch.emit(Unit)
                    updateFirstTimeLaunchPreferencesUseCase(false)
```

Reply:
Nice suggestion. @Wadeewee  👍🏻

Reply:
@Wadeewee After retrying your approach, it won't show the toast properly. So I think let's do it as-is. 🙏🏻

Reply:
@Wadeewee We can't declare the `MutableStateFlow` without the initial value, right? 😢 So this would emit to the `HomeScreen` by default. We can't use the Unit.

<img width="521" alt="Screenshot 2566-01-26 at 5 13 36 PM" src="https://user-images.githubusercontent.com/12026942/214810909-d1e26eb7-3f13-4e48-b20c-8992c56baef5.png">

Reply:
@Wadeewee Unfortunately, I tried, but it doesn't work. So let me keep the Boolean 🙏🏻

Reply:
> I tried, but it doesn't work

@lydiasama, Not sure, Is this you stuck: 

By using `MutableStateFlow` and giving the `initial value` as `Unit` It will doesn't work because whenever we enter the screen, it will trigger the `default` value `immediately`. So, we need to use `MutableSharedFlow` instead to let it trigger only when we want to emit it. The case is like when we emit ` _navigator` or `_error`.

If not this ☝️ Please let me know what are you stuck why it doesn't work 🙏

Reply:
@Wadeewee Yes, I tried with the `MutibleSharedFlow`, but it doesn't work. Could you please try on your side if I missed something?

Reply:
@lydiasama May I know do you still keep `firstTimeLaunch` in `HomeViewState` by using `MutibleSharedFlow` 🤔

Reply:
Could you please try this 🙏🏼 

```
// HomeViewModel
@HiltViewModel
class HomeViewModel @Inject constructor(
    private val getModelsUseCase: GetModelsUseCase,
    private val getFirstTimeLaunchPreferencesUseCase: GetFirstTimeLaunchPreferencesUseCase,
    private val updateFirstTimeLaunchPreferencesUseCase: UpdateFirstTimeLaunchPreferencesUseCase,
    dispatchers: DispatchersProvider
) : BaseViewModel(dispatchers) {

    private val _uiModels = MutableStateFlow<List<UiModel>>(emptyList())
    val uiModels: StateFlow<List<UiModel>>
        get() = _uiModels

    private val _firstTimeLaunch = MutableSharedFlow<Unit>()
    val firstTimeLaunch: SharedFlow<Unit>
        get() = _firstTimeLaunch

    init {
        execute {
            val getModelsFlow = getModelsUseCase()
            val getFirstTimeLaunchPreferencesFlow = getFirstTimeLaunchPreferencesUseCase()

            getModelsFlow.combine(getFirstTimeLaunchPreferencesFlow) { uiModels, firstTimeLaunch ->
                _uiModels.emit(uiModels.toUiModels())

                if (firstTimeLaunch) {
                    _firstTimeLaunch.emit(Unit)
                    updateFirstTimeLaunchPreferencesUseCase(false)
                }
            }.onStart {
                showLoading()
            }.onCompletion {
                hideLoading()
            }.catch {
                _error.emit(it)
            }.collect()
        }
    }

    fun navigateToSecond(uiModel: UiModel) {
        execute { _navigator.emit(AppDestination.Second.buildDestination(uiModel.id)) }
    }
}
```

```
// HomeScreen
@Composable
fun HomeScreen(
    viewModel: HomeViewModel = hiltViewModel(),
    navigator: (destination: AppDestination) -> Unit
) {
    val showLoading: IsLoading by viewModel.showLoading.collectAsState()
    val uiModels: List<UiModel> by viewModel.uiModels.collectAsState()
//    val firstTimeLaunch: Boolean by viewModel.firstTimeLaunch.collectAsState()

    val context = LocalContext.current
    LaunchedEffect(viewModel.error) {
        viewModel.error.collect { error ->
            val message = error.userReadableMessage(context)
            Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
        }
    }
    LaunchedEffect(viewModel.firstTimeLaunch) {
//        delay(1000)
        viewModel.firstTimeLaunch.collect {
            Toast.makeText(context, "This is the first time launch", Toast.LENGTH_SHORT).show()
        }
//        if (firstTimeLaunch) {
//            Toast.makeText(context, "This is the first time launch", Toast.LENGTH_SHORT).show()
//        }
    }
    LaunchedEffect(viewModel.navigator) {
        viewModel.navigator.collect { destination -> navigator(destination) }
    }

    CameraPermission()

    HomeScreenContent(
        uiModels = uiModels,
        showLoading = showLoading,
        onItemClick = viewModel::navigateToSecond
    )
}
```
Thanks @Wadeewee for the help 🙏🏼

Reply:
@chornerman I tried, but it doesn't show the toast message at all. Can you show the toast message by this code, I'm curious?

Reply:
@lydiasama, I can confirm it works, as I have seen when he debugs. @chornerman, Do you still have the code on your side? If it's still there so, could you please test again and screen record for her 🙇‍♀️

Reply:
@Wadeewee @chornerman Please show the record. 🙏🏻 Because I did what exactly you guys did, but it didn't work as we expected.

Reply:
@lydiasama @Wadeewee Here it is 🙏🏼 


https://user-images.githubusercontent.com/13492460/215442129-9f95edd1-d18d-47ec-b50f-e41c724b00ea.mp4

Reply:
@chornerman Thanks Chorn. It might be because my Samsung has a permission popup at the bottom, and the toast message shows on that period. 😢

Reply:
@lydiasama That is weird 🤔 I tested it on Samsung (Android 12) and it seems fine

https://user-images.githubusercontent.com/13492460/215448038-75a290b4-d586-41bb-93bf-74ad662ccd29.mp4

ANW, if this approach is not working on some devices, I'm okay with using boolean 🙌🏼